pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -936,9 +936,31 @@ func findRule(rules []network.LoadBalancingRule, rule network.LoadBalancingRule)
 
 func findSecurityRule(rules []network.SecurityRule, rule network.SecurityRule) bool {
 	for _, existingRule := range rules {
-		if strings.EqualFold(*existingRule.Name, *rule.Name) {
-			return true
+		if !strings.EqualFold(*existingRule.Name, *rule.Name) {
```

Comment:
would `reflect.DeepEqual()` help?

Reply:
I don't think DeepEqual() is helpful here as we are comparing two strings?
`Other values - numbers, bools, strings, and channels - are deeply equal if they are equal using Go's == operator.`

Reply:
so in fact probably `EqualFold` is better- `EqualFold reports whether s and t, interpreted as UTF-8 strings, are equal under Unicode case-folding.`

Reply:
I think DeepEqual is overkill. There are things that don't need to specified and really *can't* be reconciled, like the rule priority. I'll forgo comment on whether other fields should be included. Since we don't set them, I'm inclined to suggest that they can be added to the comparison in the future, just as is being done with this PR as a followup to the initial fix for sourceAddressPrefix.

Reply:
I agree. This PR should focus on finding correct differences based on what we already set in security rule.